rq/worker.py

```diff
@@ -791,6 +792,17 @@ def main_work_horse(self, job, queue):
         # contrast to the regular sys.exit()
         os._exit(0)
 
+    def main_work_horse(self, job, queue):
```

Comment:
I prefer not to have another method for the main work horse. Please wrap *only* the `perform_job()` part into a try/except block. If other parts like `random.seed()`, setting up signals and other parts fail, they should fail loudly because those are non job execution errors that need to be handled differently.

Reply:
done